src/services/dashboards/dashboard-customize/modules/DashboardWidgetInputForm.vue

```diff
@@ -212,13 +201,29 @@ export default defineComponent<Props>({
             if (Array.isArray(selectedItem)) return !!selectedItem.length;
             return selectedItem && !isEmpty(selectedItem);
         };
+        const getDashboardVariablesSchema = (propertySchema: JsonSchema['properties']) => {
+            const enabledVariables = Object.entries(dashboardDetailState.variablesSchema.properties)
+                .filter(([, d]) => {
+                    if (!d.use) return false;
+                    const variableType = d.selection_type === 'MULTI' ? 'array' : 'string';
+                    return propertySchema.type === variableType;
+                });
+            const _enum = enabledVariables.map(([key]) => key);
+            return {
+                type: 'string',
+                enum: _enum.length ? _enum : [null],
```

Comment:
description: ajv occurs error when enum is empty, so I set [null]. ([issue](https://github.com/ajv-validator/ajv/issues/824#issuecomment-723645085))